frameworks/hdfs/tests/config.py

```diff
@@ -74,7 +75,23 @@ def run_hdfs_command(svc_name, command):
     return rc, output
 
 
-def check_healthy(count=DEFAULT_TASK_COUNT):
-    sdk_plan.wait_for_completed_deployment(PACKAGE_NAME, timeout_seconds=25 * 60)
-    sdk_plan.wait_for_completed_recovery(PACKAGE_NAME, timeout_seconds=25 * 60)
-    sdk_tasks.check_running(PACKAGE_NAME, count)
+def check_healthy(service_name=PACKAGE_NAME, count=DEFAULT_TASK_COUNT, recovery_expected=False):
+    sdk_plan.wait_for_completed_deployment(service_name, timeout_seconds=25 * 60)
+    if recovery_expected:
+        # TODO(elezar): See INFINITY-2109 where we need to better handle recovery health checks
+        sdk_plan.wait_for_kicked_off_recovery(service_name, timeout_seconds=25 * 60)
+    sdk_plan.wait_for_completed_recovery(service_name, timeout_seconds=25 * 60)
+    sdk_tasks.check_running(service_name, count)
+
+
+def expect_recovery(service_name=PACKAGE_NAME):
+    # TODO(elezar, nima) check_healthy also check for complete deployment, and this should not
+    # affect the state of recovery.
+    check_healthy(service_name=service_name, count=DEFAULT_TASK_COUNT, recovery_expected=True)
+
+
+def get_pod_type_instances(pod_type_prefix, service_name=FOLDERED_SERVICE_NAME):
+    pod_types = sdk_cmd.convert_output_to_list(
```

Comment:
Oh totally missed this. It looks like the input in that example wasn't correctly formatted.
```
Python 2.7.10 (default, Feb  7 2017, 00:08:15)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import ast
>>> pods = """[
...   "data-0",
...   "data-1",
...   "data-2",
...   "journal-0",
...   "journal-1",
...   "journal-2",
...   "name-0",
...   "name-1"
... ]"""
>>> ast.literal_eval(pods)
['data-0', 'data-1', 'data-2', 'journal-0', 'journal-1', 'journal-2', 'name-0', 'name-1']
```

```
Python 3.6.1 (v3.6.1:69c0db5050, Mar 21 2017, 01:21:04)
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import ast
>>> pods = """[
...   "data-0",
...   "data-1",
...   "data-2",
...   "journal-0",
...   "journal-1",
...   "journal-2",
...   "name-0",
...   "name-1"
... ]"""
>>> ast.literal_eval(pods)
['data-0', 'data-1', 'data-2', 'journal-0', 'journal-1', 'journal-2', 'name-0', 'name-1']
```